core/base-service/legacy-request-handler.js

```diff
@@ -4,18 +4,18 @@
 const domain = require('domain')
 const request = require('request')
 const queryString = require('query-string')
-const log = require('../core/server/log')
-const analytics = require('../core/server/analytics')
-const LruCache = require('../gh-badges/lib/lru-cache')
-const makeBadge = require('../gh-badges/lib/make-badge')
+const log = require('../server/log')
+const analytics = require('../server/analytics')
+const LruCache = require('../../gh-badges/lib/lru-cache')
+const makeBadge = require('../../gh-badges/lib/make-badge')
 const {
   Inaccessible,
   InvalidResponse,
   ShieldsRuntimeError,
-} = require('../services/errors')
-const { setCacheHeaders } = require('../services/cache-headers')
-const { makeBadgeData: getBadgeData } = require('./badge-data')
-const { makeSend } = require('./result-sender')
+} = require('../../services/errors')
```

Comment:
I want to be careful to avoid circular imports, though since this is only used from `core/server/server.js` I think that would be fine.

Reply:
ah! yeah probably better safe than sorry. I'm still hoping that the legacy stuff will all be refactored soon

Reply:
I hope so! I mean, I think a new version of this functionality (i.e. caching and timeout) could be written for the modern services. I don't think it will be worth it, though, to refactor the legacy services not to use this. So we'll have those to deal with, anyway. We still have 75 files with legacy services…

I was talking with @jankeromnes about doing a distributed hack day to implement new badges, though maybe it would make more sense to get people to try to duck in and refactor. I feel like once someone has done one or two of these, they can burn through them pretty effectively.

Reply:
The method in BaseService which adds the route to camp is pretty gnarly. You'll probably recognize its structure from the legacy services you've worked on:

https://github.com/badges/shields/blob/aac2a4d5ad5f1a1494e1082a6da2a70e358643c0/services/base.js#L490-L518

It's really hard to follow. There's a call to `handleRequest` near the top which is defined in `legacy-request-handler`. Code in there delegates back to the `handler` callback that's defined here. That invokes the service, which will eventually pass control back to the caching request via `await this._request()`. (That invokes `request.isPromise()` from the `request` variable being injected here.) When the service is done it eventually invokes `sendBadge`, passing control back to code in `legacy-request-handler`.